### DISCUSSION TITLE
Timing issue

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#5484] [1. Help]
We have a Blade component for autonumeric. The input has an event listener:

```js
x-on:change="value = autonumeric.getNumber()"
```

And the component has a watcher to process it. `value` is entangled with Livewire. All of this works. 

There are situations where a user changes the input and, while keeping focus on the input, clicks a button that calls a method on the same Livewire component as the input is wired to. As we are using `x-on:change`, the data from the input field only gets send when the input loses focus. This of course happens when the user clicks the button.

This results in two requests to the Livewire component at the same time. The method that's wired to the button needs all input fields to be up to date and it requires the dom diffing to be finished (I won't get into details).

Changing the input recalculates some numbers on the frontend. We do not want to lose this functionality. This means deferring the `wire:model` is not an option for us.

What would be a solution for this?

**Comment by joshhanley** [comment]
@ju5t I think I understand what you're saying, but any chance you could mock up a demo on https://laravelplayground.com/ so we can have a look? Livewire should batch the request if the updates are within <5ms of each other, so should be working ok, I think?

**Comment by ju5t** [comment]
Sorry! I found a solution for it and forget to reply. 

I switched:

```js
x-on:change="value = autonumeric.getNumber()"
```

to

```js
x-on:input.debounce.500ms="value = autonumeric.getNumber()"
```

This can still cause timing issues, but it's less likely. I initially tried to switch it to `x-on:input` only and handle the debounce in Livewire (as that's what we prefer), but for some reason that wouldn't work. Every change would immediately trigger a call to Livewire and that's a bit heavy :)

This might be a bug (?) but I'm not sure if it's necessary to visit this while v3 is in the works. If you do want to have a look at it let me know.

**Reply by joshhanley** [reply]
@ju5t no worries! Probably not at the moment, but if it's still an issue on V3, lets have a look at it and see if we can fix it. V3 will have better request handling so it may help?
